src/lambda/routes/invocations/invocationsRoute.js

```diff
@@ -45,9 +45,10 @@ export default function invocationsRoute(lambda, options, v3Utils) {
       let statusCode = 200
       let functionError = null
       if (invokeResults) {
-        resultPayload = invokeResults.Payload || ''
-        statusCode = invokeResults.StatusCode || 200
-        functionError = invokeResults.FunctionError || null
+        let isPayloadDefined = typeof invokeResults.Payload !== 'undefined';
```

Comment:
Let's put it as `invokeResults.Payload != null`

Reply:
The ```null``` is also a valid response type. Why it has to be replaced by ```''```?

Reply:
Ok, so let's maybe clarify, what can end as payload. Shouldn't it be _string_ in all cases?

Reply:
Yes. But, ```'null'``` and ```''``` are two different strings.

Reply:
I meant `null` not `'null'`

If payload can only be string, there's nothing to change here, as only empty string is _falsy_, `'0'` is a truthy value and is definitely not replaced with an empty string here

Reply:
The ```invokeResults.Payload``` contains what the Lambda handler returns:
https://github.com/dherault/serverless-offline/blob/4fc52c40a28668b44de5347cc1b2d816f48e3f82/src/lambda/routes/invocations/InvocationsController.js#L102-L105
Regarding to Hapi source code, ```h.response()``` replaces ```undefined``` values to ```null``` anyway:
https://github.com/hapijs/hapi/blob/64c40d815bc06969ce00f2f87ecc4869d64bbdb5/lib/response.js#L94-L98
_(so we can pass even `undefined` to `h.respose()`)_

So, let's check some edge-cases (one of them I met in a real-world scenario):
1. Lambda returns ```0```:
    - Old code coverts it to ```''```, and Hapi responds with ```''``` _**(incorrect)**_
    - New code preserves it, and Hapi responds with ```'0'``` _(can be parsed back to ```0```)_
2. Lambda returns ```null```:
    - Old code coverts it to ```''```, and Hapi responds with ```''``` **_(incorrect)_**
    - New code preserves it, and Hapi responds with ```'null'```  _(can be parsed back to ```null```)_
3. Lambda returns ```false```:
    - Old code coverts it to ```''```, and Hapi responds with ```''``` **_(incorrect)_**
    - New code preserves it, and Hapi responds with ```'false'``` _(can be parsed back to ```false```)_
4. Lambda returns ```undefined```:
    - Old code coverts it to ```''```, and Hapi responds with ```''``` **_(correct)_**
    - New code behaves is the same way.

Reply:
>The invokeResults.Payload contains what the Lambda handler returns:

In real world (non-emulated lambda invocation), I assume it'll always be string (as we get the value over the network), right ?

Now, how it exactly works here (?) Will we get any non-string value as returned by handler directly. If that's the case, then a good adjustment would be to ensure this value gets stringified before being returned, to well emulate the real world behavior

Reply:
> In real world (non-emulated lambda invocation), I assume it'll always be string (as we get the value over the network), right ?

No. I case of AWS AppSync, Lambda Resolver can return ```0``` or ```null``` and it will be added to resulting graph as is (and actually in works in the cloud). But the `serverless-offline` (it is used by `serverless-appsync-simulator`) replaces `0` by `''`, which breaks the graph types validation.

P.S. You've received an external contribution to your package, which fixes annoying bug, and you decided to argue instead of saying "thank you". Oh...

Reply:
>P.S. You've received an external contribution to your package, which fixes annoying bug, and you decided to argue instead of saying "thank you". Oh...

@trsrm sorry you take it that way. I just don't want to take any PR blindly, without fully understanding that indeed it's the right fix. Thanks to that you also get to work with higher quality product

Reply:
That could be my fault, as I didn't have enough time to describe the context in details...
We are building pure serverless application using `serverless` and `serverless-appsync`. Now, we are working to add the offline capabilities, and noticed that some requests were failing on local environment. Then, I had to debug everything step-by-step, and found this issue. 
P.S. Next time, I will add more details.